test/dataframe.jl

```diff
@@ -2066,12 +2071,30 @@ end
                               C=[fill(missing, 6); 7:9],
                               source=[-4, -4, -4, -3, -3, -3, -2, -2, -2])
         @test res.source isa CategoricalVector
+
+        res = reduce(vcat, [df1, df2, df3, df4], cols=:union, source=col => categorical(-4:-1))
+        @test res ≅ DataFrame(A=1:9, B=[1:6; fill(missing, 3)],
+                              C=[fill(missing, 6); 7:9],
+                              source=[-4, -4, -4, -3, -3, -3, -2, -2, -2])
+        @test res.source isa CategoricalVector
```

Comment:
Why not just do `isequal_coltyped(res, reduce(vcat, [df1, df2, df3, df4], cols=:union, source=col => categorical(-4:-1)))` in the block above?

Reply:
fixed - it was a leftover of old tests.